database/src/lib.rs

```diff
@@ -344,6 +354,14 @@ pub trait ScyllaStorageManager {
             .known_node(scylla_url)
             .default_execution_profile_handle(scylla_execution_profile_handle);
 
+        if let Some(scylla_dc_host_filter) = scylla_dc_host_filter {
+            session = session.host_filter(std::sync::Arc::new(
+                scylla::transport::host_filter::DcHostFilter::new(
```

Comment:
Oh, so the problem is not with establishment of TCP sessions but with Scylla driver using ALL datacenters (including the ones across ocean)! This explains better 150ms write duration.

Reply:
Yep, your experiment and results led me to this conclusion. So I've implemented this and confirmed that was the root cause. Now it seems obvious.